### PR TITLE
Add country code logic and UI to test app

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the UWP implementation?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Add logic and UI for setting, enabling and disabling country code via AppCenter.SetCountryCode to WPF\WinForms test app

form

form2

## Related PRs or issues

[AB#63416](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/63416)

### CODE REVIEW

File: Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs [hunk 1]
```diff
@@ -95,6 +95,38 @@ private void TrackEvent_Click(object sender, RoutedEventArgs e)
             Analytics.TrackEvent(name, propertiesDictionary);
         }
 
+        private void countryCodeEnabled_Checked(object sender, RoutedEventArgs e)
```
Comment (guperrot) [inline Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs]: rename to upper case C, CountryCodeEnabled

File: Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs [hunk 2]
```diff
@@ -95,6 +95,38 @@ private void TrackEvent_Click(object sender, RoutedEventArgs e)
...
+                else
+                {
+                    lCountryCode.Text = RegionInfo.CurrentRegion.TwoLetterISORegionName;
+                    AppCenter.SetCountryCode(lCountryCode.Text);
+                }
+                spCountryCode.IsEnabled = cbEnableCountryCode.IsChecked.Value;
+            }
+        }
+
+        private void countryCodeSave_ClickListener(object sender, RoutedEventArgs e)
```
Comment (guperrot) [inline Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs]: same here

File: Apps/Contoso.WinForms.Puppet/MainForm.cs [hunk 3]
```diff
@@ -167,5 +168,34 @@ public static async Task DoStuffInBackground()
         }
 
         #endregion
+
+        private void CountruCodeEnable_CheckedChanged(object sender, EventArgs e)
```
Comment (guperrot) [inline Apps/Contoso.WinForms.Puppet/MainForm.cs]: typo

File: Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs [hunk 4]
```diff
@@ -95,6 +95,38 @@ private void TrackEvent_Click(object sender, RoutedEventArgs e)
...
+                    AppCenter.SetCountryCode(lCountryCode.Text);
+                }
+                spCountryCode.IsEnabled = cbEnableCountryCode.IsChecked.Value;
+            }
+        }
+
+        private void CountryCodeSave_ClickListener(object sender, RoutedEventArgs e)
+        {
+            // Checked that input country code is valid.
+            bool isValidCountryCode = CultureInfo.GetCultures(CultureTypes.SpecificCultures)
```
Comment (achocron) [inline Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs]: Why do we test if the code is valid? Might be a valid test scenario to set an invalid country code.

File: Apps/Contoso.WinForms.Puppet/MainForm.cs [hunk 5]
```diff
@@ -167,5 +168,34 @@ public static async Task DoStuffInBackground()
...
+            {
+                lCountryCode.Text = RegionInfo.CurrentRegion.TwoLetterISORegionName;
+                AppCenter.SetCountryCode(lCountryCode.Text);
+            }
+            gbCountryCode.Enabled = cbCountryCode.Checked;
+        }
+
+        private void BtnSave_ClickListener(object sender, EventArgs e)
+        {
+            // Checked that input country code is valid.
```
Comment (Jamminroot) [inline Apps/Contoso.WinForms.Puppet/MainForm.cs]:
```suggestion
            // Check that input country code is valid.
```

File: Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs [hunk 6]
```diff
@@ -95,6 +95,38 @@ private void TrackEvent_Click(object sender, RoutedEventArgs e)
...
+                    lCountryCode.Text = RegionInfo.CurrentRegion.TwoLetterISORegionName;
+                    AppCenter.SetCountryCode(lCountryCode.Text);
+                }
+                spCountryCode.IsEnabled = cbEnableCountryCode.IsChecked.Value;
+            }
+        }
+
+        private void CountryCodeSave_ClickListener(object sender, RoutedEventArgs e)
+        {
+            // Checked that input country code is valid.
```
Comment (Jamminroot) [inline Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs]:
```suggestion
            // Check that input country code is valid.
```

File: Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs [hunk 7]
```diff
@@ -95,6 +95,38 @@ private void TrackEvent_Click(object sender, RoutedEventArgs e)
             Analytics.TrackEvent(name, propertiesDictionary);
         }
 
+        private void CountryCodeEnabled_Checked(object sender, RoutedEventArgs e)
+        {
+            if (cbEnableCountryCode.IsChecked.HasValue)
```
Comment (Jamminroot) [inline Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs]: I think this should be refactored to make it more readable

File: Apps/Contoso.WinForms.Puppet/MainForm.Designer.cs [hunk 8]
```diff
@@ -539,6 +603,11 @@ private void InitializeComponent()
         private System.Windows.Forms.Button CrashWithAggregateException;
         private System.Windows.Forms.Button CrashWithNullReference;
         private System.Windows.Forms.Button CrashInsideAsyncTask;
+        private System.Windows.Forms.GroupBox gbCountryCode;
+        private System.Windows.Forms.Label label2;
+        private System.Windows.Forms.Button button1;
```
Comment (Jamminroot) [inline Apps/Contoso.WinForms.Puppet/MainForm.Designer.cs]: I think we can change names of `button1` and `label2` to describe what they are for in short:)

File: Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs (hunk 7 above)
Comment (Jamminroot) [reply Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs]: Not resolved yet, reactivating the comment.
We can use guardian pattern here to reduce nesting

File: Apps/Contoso.WPF.Puppet/MainWindow.xaml [hunk 9]
```diff
@@ -93,7 +93,21 @@
...
             </TabItem>
-            <TabItem Header="Others"/>
+            <TabItem Header="Others">
+                <StackPanel>
+                    <CheckBox Name="cbEnableCountryCode" Content="Country Code Enabled" Margin="0,10,0,10" Checked="CountryCodeEnabled_Checked" Unchecked="CountryCodeEnabled_Checked"/>
+                    <GroupBox Header="Country code">
+                        <StackPanel IsEnabled="False" Name="spCountryCode">
+                            <DockPanel LastChildFill="True">
+                                <Label Name="tbCountryCode">Country code</Label>
+                                <TextBox Name="lCountryCode"  VerticalAlignment="Center"></TextBox>
```
Comment (annakocheshkova) [inline Apps/Contoso.WPF.Puppet/MainWindow.xaml]: lCountryCode = countryCodeText

File: Apps/Contoso.WPF.Puppet/MainWindow.xaml [hunk 10]
```diff
@@ -93,7 +93,21 @@
...
                 </StackPanel>
             </TabItem>
-            <TabItem Header="Others"/>
+            <TabItem Header="Others">
+                <StackPanel>
+                    <CheckBox Name="cbEnableCountryCode" Content="Country Code Enabled" Margin="0,10,0,10" Checked="CountryCodeEnabled_Checked" Unchecked="CountryCodeEnabled_Checked"/>
+                    <GroupBox Header="Country code">
+                        <StackPanel IsEnabled="False" Name="spCountryCode">
+                            <DockPanel LastChildFill="True">
+                                <Label Name="tbCountryCode">Country code</Label>
```
Comment (annakocheshkova) [inline Apps/Contoso.WPF.Puppet/MainWindow.xaml]: tbCountryCode = countryCodeLabel

File: Apps/Contoso.WPF.Puppet/MainWindow.xaml [hunk 11]
```diff
@@ -93,7 +93,21 @@
                     </GroupBox>
                 </StackPanel>
             </TabItem>
-            <TabItem Header="Others"/>
+            <TabItem Header="Others">
+                <StackPanel>
+                    <CheckBox Name="cbEnableCountryCode" Content="Country Code Enabled" Margin="0,10,0,10" Checked="CountryCodeEnabled_Checked" Unchecked="CountryCodeEnabled_Checked"/>
+                    <GroupBox Header="Country code">
+                        <StackPanel IsEnabled="False" Name="spCountryCode">
```
Comment (annakocheshkova) [inline Apps/Contoso.WPF.Puppet/MainWindow.xaml]: spCountryCode = countryCodePanel

File: Apps/Contoso.WPF.Puppet/MainWindow.xaml [hunk 12]
```diff
@@ -93,7 +93,21 @@
                     </GroupBox>
                 </StackPanel>
             </TabItem>
-            <TabItem Header="Others"/>
+            <TabItem Header="Others">
+                <StackPanel>
+                    <CheckBox Name="cbEnableCountryCode" Content="Country Code Enabled" Margin="0,10,0,10" Checked="CountryCodeEnabled_Checked" Unchecked="CountryCodeEnabled_Checked"/>
```
Comment (annakocheshkova) [inline Apps/Contoso.WPF.Puppet/MainWindow.xaml]: cbEnableCountryCode = countryCodeEnabledCheckbox

File: Apps/Contoso.WPF.Puppet/MainWindow.xaml [hunk 13]
```diff
@@ -93,7 +93,21 @@
...
+                    <GroupBox Header="Country code">
+                        <StackPanel IsEnabled="False" Name="spCountryCode">
+                            <DockPanel LastChildFill="True">
+                                <Label Name="tbCountryCode">Country code</Label>
+                                <TextBox Name="lCountryCode"  VerticalAlignment="Center"></TextBox>
+                            </DockPanel>
+                            <Button Name="btnSaveCountryCode" Click="CountryCodeSave_ClickListener">Save</Button>
```
Comment (annakocheshkova) [inline Apps/Contoso.WPF.Puppet/MainWindow.xaml]: btnSaveCountryCode = saveCountryCodeButton

File: Apps/Contoso.WinForms.Puppet/MainForm.cs [hunk 14]
```diff
@@ -167,5 +168,25 @@ public static async Task DoStuffInBackground()
         }
 
         #endregion
+
+        private void CountryCodeEnable_CheckedChanged(object sender, EventArgs e)
+        {
+            if (!cbCountryCode.Checked)
```
Comment (annakocheshkova) [inline Apps/Contoso.WinForms.Puppet/MainForm.cs]: Same for the names

File: Apps/Contoso.WPF.Puppet/MainWindow.xaml [hunk 15]
```diff
@@ -93,7 +93,21 @@
                     </GroupBox>
                 </StackPanel>
             </TabItem>
-            <TabItem Header="Others"/>
+            <TabItem Header="Others">
+                <StackPanel>
+                    <CheckBox Name="countryCodeEnabledCheckbox" Content="Country Code Enabled" Margin="0,10,0,10" Checked="CountryCodeEnabled_Checked" Unchecked="CountryCodeEnabled_Checked"/>
```
Comment (guperrot) [inline Apps/Contoso.WPF.Puppet/MainWindow.xaml]: There is a warning when you use camelCase. We fixed that everywhere in the apps in previous PRs.

Please fix:

Capture

File: Apps/Contoso.WinForms.Puppet/MainForm.cs [hunk 16]
```diff
@@ -167,5 +168,25 @@ public static async Task DoStuffInBackground()
         }
 
         #endregion
+
+        private void CountryCodeEnable_CheckedChanged(object sender, EventArgs e)
+        {
+            if (!countryCodeEnabledCheckbox.Checked)
```
Comment (guperrot) [inline Apps/Contoso.WinForms.Puppet/MainForm.cs]: We changed all those to PascalCase due to a previous PR feedback.

File: Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs [hunk 17]
```diff
@@ -95,6 +95,32 @@ private void TrackEvent_Click(object sender, RoutedEventArgs e)
...
+               CountryCodeText.Text = "";
+                AppCenter.SetCountryCode(null);
+            }
+            else
+            {
+                CountryCodeText.Text = RegionInfo.CurrentRegion.TwoLetterISORegionName;
+                AppCenter.SetCountryCode(CountryCodeText.Text);
+            }
+            CountryCodePanel.IsEnabled = CountryCodeEnableCheckbox.IsChecked.Value;
+
```
Comment (MatkovIvan) [inline Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs]: Extra line. Please format.

File: Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs [hunk 18]
```diff
@@ -95,6 +95,31 @@ private void TrackEvent_Click(object sender, RoutedEventArgs e)
             Analytics.TrackEvent(name, propertiesDictionary);
         }
 
+        private void CountryCodeEnabled_Checked(object sender, RoutedEventArgs e)
+        {
+            if (!CountryCodeEnableCheckbox.IsChecked.HasValue)
+            {
+                return;
+            }
+
```
Comment (MatkovIvan) [inline Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs]:
```suggestion
```

File: Apps/Contoso.WPF.Puppet/MainWindow.xaml [hunk 19]
```diff
@@ -93,7 +93,21 @@
...
                 </StackPanel>
             </TabItem>
-            <TabItem Header="Others"/>
+            <TabItem Header="Others">
+                <StackPanel>
+                    <CheckBox Name="CountryCodeEnableCheckbox" Content="Country Code Enabled" Margin="0,10,0,10" Checked="CountryCodeEnabled_Checked" Unchecked="CountryCodeEnabled_Checked"/>
+                    <GroupBox Header="Country code">
+                        <StackPanel IsEnabled="False" Name="CountryCodePanel">
+                            <DockPanel LastChildFill="True">
+                                <Label Name="CountryCodeLable">Country code</Label>
```
Comment (guperrot) [inline Apps/Contoso.WPF.Puppet/MainWindow.xaml]: Typo
```suggestion
                                <Label Name="CountryCodeLabel">Country code</Label>
```

File: Apps/Contoso.WinForms.Puppet/MainForm.Designer.cs [hunk 20]
```diff
@@ -60,6 +60,11 @@ private void InitializeComponent()
             this.CrashWithTestException = new System.Windows.Forms.Button();
             this.CrashesEnabled = new System.Windows.Forms.CheckBox();
             this.OthersTab = new System.Windows.Forms.TabPage();
+            this.CountryCodeGroup = new System.Windows.Forms.GroupBox();
+            this.label2 = new System.Windows.Forms.Label();
```
Comment (guperrot) [inline Apps/Contoso.WinForms.Puppet/MainForm.Designer.cs]: Rename label like the other ones

**Pull request merged**